### PR TITLE
Minor updates to PN counter and CRDT replication documentation

### PR DESCRIPTION
**Pull request by mmedenjak**

### CODE REVIEW

File: src/CRDTPNCounter.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
-
-
 ## CRDT PN-Counter
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: ## PN Counter

File: src/CRDTPNCounter.md [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+Hazelcast offers a lightweight CRDT PN-Counter (Positive-Negative Counter) implementation where each hazelcast instance can increment and decrement the counter value and these updates are propagated to all replicas. Only a hazelcast member can store state for a counter which means that counter method invocations performed on a hazelcast member are usually local (depending on the configured replica count). If there is no member failure, it is guaranteed that each replica sees the final value of the counter eventually. Counter's state converges with each update and all CRDT replicas that can communicate to each other will eventually have the same state.
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: ...where each Hazelcast instance can.....Only a Hazelcast member can store.....performed on a Hazelcast member are......

File: src/CRDTPNCounter.md (hunk 1 above)
Comment (Serdaro) [inline src/CRDTPNCounter.md]: ## PN Counter

File: src/CRDTPNCounter.md (hunk 2 above)
Comment (Serdaro) [inline src/CRDTPNCounter.md]: ...a lightweight CRDT PN counter (Positive-........

... Only a Hazelcast member can store.....

...performed on a Hazelcast member are usually...

File: src/CRDTPNCounter.md [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+Hazelcast offers a lightweight CRDT PN-Counter (Positive-Negative Counter) implementation where each hazelcast instance can increment and decrement the counter value and these updates are propagated to all replicas. Only a hazelcast member can store state for a counter which means that counter method invocations performed on a hazelcast member are usually local (depending on the configured replica count). If there is no member failure, it is guaranteed that each replica sees the final value of the counter eventually. Counter's state converges with each update and all CRDT replicas that can communicate to each other will eventually have the same state. 
 
-Using the CRDT PN-Counter, you can get a distributed counter, increment and decrement it, and query its value with RYW (read-your-writes) and monotonic reads.
+Using the CRDT PN-Counter, you can get a distributed counter, increment and decrement it, and query its value with RYW (read-your-writes) and monotonic reads. The implementation borrows most methods from the `AtomicLong` which should be familiar in most cases and easily interchangeable in existing code.
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: ....in the existing code.

File: src/CRDTPNCounter.md [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
-??? Maybe some real-life use case examples can be mentioned here ???
+Some examples of PN counter use are:
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: Some example usages of PN counter are:

File: src/CRDTPNCounter.md [hunk 5]
```diff
@@ -30,7 +32,7 @@ final HazelcastInstance instance = Hazelcast.newHazelcastInstance();
         final long value = counter.get();
 ```
 
-??? What does this example do ???
+This code snippet will create an instance of a PN counter, increment it by 5 and retrieve the value.
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: This code snippets creates an....increments it by 5 and retrieves..

File: src/CRDTPNCounter.md [hunk 6]
```diff
@@ -30,7 +32,7 @@ final HazelcastInstance instance = Hazelcast.newHazelcastInstance();
...
 
 Please refer to the [PN-Counter Javadoc](http://docs.hazelcast.org/docs/3.10/javadoc/com/hazelcast/crdt/pncounter/PNCounter.java) for its API documentation.
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: Remove the link and its text for now, will be added before the release.

File: src/CRDTPNCounter.md [hunk 7]
```diff
@@ -43,13 +45,50 @@ Following is an example declarative configuration snippet:
...
-    <quorum-name>quorumname</quorum-name>
     <statistics-enabled>true</statistics-enabled>
   </pn-counter>
 </hazelcast>
 ```
 
 - `name`: Name of your PN-Counter.
-- `replica-count`: Number of replicas on which state for this PN counter will be kept. This number applies in quiescent state, if there are currently membership changes or clusters are merging, the state may be temporarily kept on more replicas. Its default value is Integer.MAX_VALUE.
-- `quorum-name`: Name of quorum configuration that you want this PN-Counter to use.
+- `replica-count`: Number of replicas on which state for this PN counter will be kept. This number applies in quiescent state, if there are currently membership changes or clusters are merging, the state may be temporarily kept on more replicas. Its default value is Integer.MAX_VALUE. Generally, keeping the state on more replicas means that more hazelcast members will be able to perform updates locally but it also means that the PN counter state will be kept on more replicas, increasing the network traffic, decreasing the speed at which replica states converge and increasing the size of the PN counter state kept on each replica.
```
Comment (Serdaro) [inline src/CRDTPNCounter.md]: ...more Hazelcast members will be...

**Comment by mmedenjak** [comment]
@Serdaro addressed all comments

**Comment by mmedenjak** [comment]
@Serdaro looks good? Any other comments?

**Comment by Serdaro** [comment]
LGTM:)

**Pull request merged**